src/Adapter/MSTest.TestAdapter/Execution/TypeCache.cs

```diff
@@ -145,9 +145,11 @@ public IEnumerable<TestAssemblyInfo> AssemblyInfoListWithExecutableCleanupMethod
         DebugEx.Assert(testMethod != null, "test method is null");
 
         string typeName = testMethod.FullClassName;
-
-        if (!_classInfoCache.TryGetValue(typeName, out TestClassInfo? classInfo))
+        return _classInfoCache.GetOrAdd(typeName, static (typeName, tuple) =>
```

Comment:
Is this the crux of the whole PR? to avoid race condition in the info cache by using proper concurrent api rather than TryGetValue?

Reply:
Yup.